tools/tests/test_check_constitution.py

```
@@ -0,0 +1,345 @@
+    src = Path(__file__).resolve().parent.parent / "check_constitution.py"
+    dst = repo / "tools" / "check_constitution.py"
+    dst.parent.mkdir(parents=True, exist_ok=True)
+    dst.write_text(src.read_text(encoding="utf-8"), encoding="utf-8")
+    return dst
+
+
+def _exit_code(repo: Path, *extra: str) -> int:
+    """Invoke the guard as CI does. The pins above call run() in-process, which
+    cannot see main()'s return value — sabotaging it left all of them green."""
+    guard = _install_guard(repo)
+    return subprocess.run(
```

Ruff (0.16.1)

[error] `subprocess` call: check for execution of untrusted input

(S603)